### ISSUE TITLE
Handle no region in aws config

### ISSUE THREAD

**Issue by jayair** [sst/ion#125]
If you don't have a region in your AWS config, you get the following on `sst dev`:

```
$ sst dev
×  Unexpected error occurred. Please check the logs or run with --verbose for more details.
```

The `.sst/sst.log` is not helpful.

```
time=2024-03-28T16:57:32.323-04:00 level=ERROR msg="exited with error" err="operation error IoT: DescribeEndpoint, https response error StatusCode: 0, RequestID: , request send failed, Get \"https://iot..amazonaws.com/endpoint\": dial tcp: lookup iot..amazonaws.com: no such host"
```

We should handle this error better.

**Comment by jamesgibbons92** [comment]
just to add to this, this api call doesn't seem to be using the profile specified in providers.aws.profile. I was having issues running `sst dev` and it was because it was trying to use my default profile (which isn't the profile I use to deploy the sst app). so I had to specify the env var AWS_PROFILE as well as setting providers.aws.profile

I would have thought this should use provider profile, otherwise it may describe an endpoint not on the account / region you are actually deploying to.

**Comment by thdxr** [comment]
^ i just fixed this in 0.0.230

**Comment by thdxr** [comment]
iot wasn't checking region correctly - fixed in 0.0.240

i don't think region = auto is a thing so that second error is probably correct
